src/main.rs

```diff
@@ -1,23 +1,115 @@
+                    Instruction::Store(store) => self.store_var(func_name, &store.address, &store.value),
+                    _ => todo!(),
+                }
+            }
+            match &bb.term {
+                Terminator::Ret(rec) => {
+                    match rec.return_operand.as_ref().unwrap() {
+                        LocalOperand { name, ty } => todo!(),
+                        ConstantOperand(con_ref) => {
+                            match con_ref.as_ref() {
+                                Constant::Int { bits, value } => {
+                                    return Some(*value);
+                                },
+                                _ => todo!(),
+                            }
+                        },
+                        MetadataOperand => todo!(),
+                    }
+                },
+                _ => todo!(),
+            }
+        } 
+        None
+    }
+
+    fn store_var(&mut self, func_name: &String, add: &Operand, val: &Operand) {
+        let add = self.add_str(func_name, add);
+        match val {
+            LocalOperand { name, ty } => todo!(),
+            ConstantOperand(con_ref) => {
+                match con_ref.as_ref() {
+                    Constant::Int { bits, value } => {
+                        self.mem.insert(add, *value);
+                    },
+                    _ => todo!(),
+                }
+            },
+            MetadataOperand => todo!(),
+        }
+    }
+
+    fn add_str(&self, func_name: &String, add: &Operand) -> String {
+        match add {
+            LocalOperand { name, ty } => {
+                match name {
+                    Name(_) => todo!(),//return format!("let {} =", bstr),
+                    Number(n) => {
+                         return format!("{}{}", func_name, n);
+                    },
+                }
+            },
+            ConstantOperand(con_ref) => {
+                match con_ref.as_ref() {
+                    Constant::Int { bits, value } => todo!(),
+                    _ => todo!(),
+                }
+            },
+            MetadataOperand => todo!(),//{return "".to_string()},
+        }
+    }
+}
 
 fn main() {
-    let module: Module;
-    match create_module("tests/simple.bc") {
-        Ok(m) => {
-            module = m;
-            match module.get_func_by_name("main") {
-                Some(main) => interprete_llvm_ir(main),
-                None => println!("No main function"),
+    let mut bf = BFInterp::new();
+    let module = match create_module("tests/simple.bc") {
+        Ok(module) => {
+            match bf.interpret(module) {
+                Ok(_) => {},
+                Err(str) => println!("{}", str),
             }
         },
         Err(error_message) => println!("{}", error_message),
-    }
+    };
 }
 
 fn create_module(path_s: &str) -> Result<Module, String> {
     Module::from_bc_path(path_s)
-}
-
-fn interprete_llvm_ir(main_func: &Function) {
-    
-}
+}
```

Comment:
Can you try and change VSCode to add newlines to the end of files? https://stackoverflow.com/questions/44704968/visual-studio-code-insert-new-line-at-the-end-of-files